configure.ac

```diff
@@ -7,6 +7,11 @@ AH_TOP([#ifndef LIBSECP256K1_CONFIG_H])
 AH_TOP([#define LIBSECP256K1_CONFIG_H])
 AH_BOTTOM([#endif /*LIBSECP256K1_CONFIG_H*/])
 AM_INIT_AUTOMAKE([foreign subdir-objects])
+
+# Set -g if CFLAGS are not already set, which matches the default autoconf
+# behavior (see PROG_CC in the Autoconf manual) with the exception that we don't
+# set -O2 here because we set it in any case (see further down).
+: ${CFLAGS="-g"}
```

Comment:
Couldn't we get rid of this entirely then because Autoconf's default is `-g -O2`?

If your answer is, maybe keep this variant still as a branch somewhere until we have further Concept ACKs on the idea of switching to `-O2` :p

Reply:
If we remove this then if `$CFLAGS` aren't already set they will be set to `-O2 ... -O2`. The comment additionally serves as a hint to how autoconf works :) because I'd rather set `: ${CFLAGS=""}` in the future, because anything else doesn't make much sense to me.

Reply:
Okay yes. Also, autoconf sets -O2 apparently only for gcc and not for other compilers according to its manual... So we need to keep the -O2 further below anyway.